projects/packages/assets/src/class-semver.php

```diff
@@ -102,7 +102,7 @@ public static function compare( $a, $b ) {
 			if ( ctype_digit( $a ) ) {
 				if ( ctype_digit( $b ) ) {
 					if ( (int) $a !== (int) $b ) {
-						return $a - $b;
+						return (int) $a - (int) $b;
```

Comment:
It seems this whole section should be simpler. But also I'm curious why something like `version_compare` wasn't used. 🤔

Reply:
Because `version_compare` compares "php version strings", not semver version strings. They differ significantly in how prereleases are handled.

And I didn't use something like composer/semver because it didn't seem worth bringing in a whole extra package (and dealing with the possible version conflicts and such) for something that can be done in a singe small file.

You're welcome to try to make it simpler (in a separate PR, please), but I don't think you'll do much better than either like this or like
```php
$a_is_numeric = ctype_digit( $a );
$b_is_numeric = ctype_digit( $b );
if ( $a_is_numeric !== $b_is_numeric ) {
    return $a_is_numeric ? -1 : 1;
} elseif ( $a_is_numeric && (int) $a !== (int) $b ) {
    return (int) $a - (int) $b;
} elseif ( ! $a_is_numeric ) {
    $tmp = strcmp( $a, $b );
    if ( $tmp !== 0 ) {
        return $tmp;
    }
}
```
which doesn't strike me as obviously better or worse.